sdk/pinocchio/src/account_info.rs

```diff
@@ -15,6 +15,26 @@ use crate::{program_error::ProgramError, pubkey::Pubkey, ProgramResult};
 /// single top-level instruction.
 pub const MAX_PERMITTED_DATA_INCREASE: usize = 1_024 * 10;
 
+/// Represents masks for borrow state of an account.
+#[repr(u8)]
+#[derive(Clone, Copy)]
+pub enum BorrowState {
+    /// Mask to check whether an account is already borrowed.
+    ///
+    /// This will test both data and lamports borrow state.
+    Borrowed = 0b_1111_1111,
+
+    /// Mask to check whether an account is already mutably borrowed.
+    ///
+    /// This will test both data and lamports mutable borrow state.
+    MutablyBorrowed = 0b_1000_1000,
+
+    /// Mask to check whether an account is already immutably borrowed.
+    ///
+    /// This will test both data and lamports immutably borrow state.
+    ImmutablyBorrowed = 0b_0111_0111,
```

Comment:
Is this used at all? If not, probably better to just remove it

Reply:
Removed it.